src/lava/networks/gradedvecnetwork.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2022-23 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from lava.proc.graded.process import InvSqrt
+from lava.proc.graded.process import NormVecDelay
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.process import Dense
+from lava.proc.prodneuron.process import ProdNeuron
+from lava.proc.graded.process import GradedVec
+from lava.proc.lif.process import LIF
+from lava.proc.io import sink, source
+from lava.proc import embedded_io as eio
+
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from .network import AlgebraicVector, AlgebraicMatrix
+
+
+class InputVec(AlgebraicVector):
+    """InputVec
+    Simple input vector. Adds algebraic syntax to RingBuffer
```

Comment:
Please follow Python docstring conventions as specified by lava's [developer guidelines](https://lava-nc.org/developer_guide.html#docstring-format).
In turn, these follow [numpy docstring conventions](https://numpydoc.readthedocs.io/en/latest/format.html), which, in turn follow Python [PEP 257 docstring conventions](https://peps.python.org/pep-0257/).